pages/Home.vue

```diff
@@ -106,60 +107,72 @@ export default {
       loadNewsletterPopup: false,
       dummyInstaImages: [
         {
-          mobile: { url: `/assets/ig/ig01.jpg` },
-          desktop: { url: `/assets/ig/ig01.jpg` }
+          webp: { url: `/assets/ig/webp/ig01.webp` },
+          fallback: { url: `/assets/ig/jpg/ig01.jpg` }
         },
         {
-          mobile: { url: `/assets/ig/ig02.jpg` },
-          desktop: { url: `/assets/ig/ig02.jpg` }
+          webp: { url: `/assets/ig/webp/ig02.webp` },
+          fallback: { url: `/assets/ig/jpg/ig02.jpg` }
         },
         {
-          mobile: { url: `/assets/ig/ig03.jpg` },
-          desktop: { url: `/assets/ig/ig03.jpg` }
+          webp: { url: `/assets/ig/webp/ig03.webp` },
+          fallback: { url: `/assets/ig/jpg/ig03.jpg` }
         },
         {
-          mobile: { url: `/assets/ig/ig04.jpg` },
-          desktop: { url: `/assets/ig/ig04.jpg` }
+          webp: { url: `/assets/ig/webp/ig04.webp` },
+          fallback: { url: `/assets/ig/jpg/ig04.jpg` }
         },
         {
-          mobile: { url: `/assets/ig/ig05.jpg` },
-          desktop: { url: `/assets/ig/ig05.jpg` }
+          webp: { url: `/assets/ig/webp/ig05.webp` },
+          fallback: { url: `/assets/ig/jpg/ig05.jpg` }
         },
         {
-          mobile: { url: `/assets/ig/ig06.jpg` },
-          desktop: { url: `/assets/ig/ig06.jpg` }
+          webp: { url: `/assets/ig/webp/ig06.webp` },
+          fallback: { url: `/assets/ig/jpg/ig06.jpg` }
         }
-      ]
+      ],
+      newsletterImage: null,
+      banners: [],
+      heroes: []
     };
   },
   computed: {
     ...mapGetters({
       isLoggedIn: 'user/isLoggedIn',
-      heroImage: 'promoted/getHeadImage',
+      heroImages: 'promoted/getHeadImage',
       promotedOffers: 'promoted/getPromotedOffers',
       newCollection: 'homepage/getEverythingNewCollection'
     }),
     isOnline () {
       return onlineHelper.isOnline;
+    }
+  },
+  methods: {
+    showNewsletterPopup () {
+      this.loadNewsletterPopup = true;
+      this.$bus.$emit('modal-show', 'modal-newsletter');
     },
-    heroes () {
-      const hero = {
-        ...this.heroImage
-      };
-
-      return [hero, hero, hero];
+    createBanners (webpSupported) {
```

Comment:
maybe put this to seperate file as you used it two times :) and this can be used in other places in future

Reply:
I created helper